### PR TITLE
feat: Matching charts to wireframe [WEB-684]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Adapting the chart on UI Kit to match our expected wireframe

Specific changes:
- design of chart legend (replacing uPlot built-in legend)
- font family and size of text in title, legend, and axes / ticks
- background has horizontal lines, not grid
- width of lines
- removing individual point-circles (even when hovering? I can add nearest point visible on hover)

API now expects an associated Metric for the chart and MetricType for the series

Screen Shot 2023-01-19 at 3 52 15 PM

Concerns: 

You might notice what looks kind of like solid axes at x=0 y=0. But it's a vertical bar through the first point (if you move the mouse, this becomes more apparent). I also added a negative number to make it clear there's no solid x-axis, just a y=0 line

Screen Shot 2023-01-19 at 4 12 17 PM

Screen Shot 2023-01-19 at 4 12 23 PM

## Test Plan

Visit `/det/design` and scroll to the bottom to view charts section and chart grid

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by ClaireNeveu** [comment]
Point circles should be visible when hovering and they should be little diamonds.

**Comment by mapmeld** [comment]
I added diamond point on hover 🔷 to the charts. ChartGrid adds a working scale switch, and a not-working x-axis switch if you send X-Axis options (the workings of this future x-axis functionality are a big complicated; we might want to send an event back to the page and call the API for new data, rather than shuffling data around for the times to be on the x-axis)

I'm open to the X-Axis control having an enum of acceptable options (batches, time, future epochs).

Screen Shot 2023-01-20 at 9 58 57 AM

**Comment by ClaireNeveu** [comment]
I guess there's an optimization question here. Do we want to load all the data on page load to make switching faster or delay loading data with other x-axes until you select them?

If the former I would expect an API like this:
```typescript
{ chartsProps: Array<{ [axis: String]: LineChartProps }>
```
in the latter case I would expect something like this:
```typescript
{ chartsProps  : Loadable<Array<LineChartProps>>
, xAxes        : Array<string>,
, onAxisSelect : (axis: string) => void
}
```
The first case is definitely simpler since it puts all the logic into the component but I don't know enough about the volume of data we have to load on these charts to know if it would be a performance problem to pre-emptively load them.

File: webui/react/src/components/XAxisFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { Select } from 'antd';
```
Comment (ClaireNeveu) [inline webui/react/src/components/XAxisFilter.tsx:1]: Can you move this into the proper location? Should probably be `/components/kit/LineChart/XAxisFilter.tsx`

File: webui/react/src/components/kit/LineChart.tsx [hunk 2]
```diff
@@ -72,39 +94,62 @@ export const LineChart: React.FC<Props> = ({
...
+        </div>
+      )}
     </>
   );
 };
 
 interface GroupProps {
+  chartSettingsControls?: boolean;
   chartsProps: Props[];
   rowHeight?: number;
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/LineChart.tsx:137]: What is `rowHeight` for?

File: webui/react/src/components/kit/LineChart.tsx [hunk 3]
```diff
@@ -72,39 +94,62 @@ export const LineChart: React.FC<Props> = ({
...
+            </li>
+          ))}
+        </div>
+      )}
     </>
   );
 };
 
 interface GroupProps {
+  chartSettingsControls?: boolean;
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/LineChart.tsx:135]: Controls should always be visible but you probably want to automatically hide the axis control if only one axis is passed.

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]

**Comment by ClaireNeveu** [comment]
The point indicator still does not match the wireframe and the background color is wrong.
Screen Shot 2023-01-20 at 13 35 25
Screen Shot 2023-01-20 at 13 35 55

File: webui/react/src/components/XAxisFilter.tsx (hunk 1 above)
Comment (mapmeld) [reply webui/react/src/components/XAxisFilter.tsx:1]: ✅

File: webui/react/src/components/kit/LineChart.tsx (hunk 3 above)
Comment (mapmeld) [reply webui/react/src/components/kit/LineChart.tsx:135]: ✅

File: webui/react/src/components/kit/LineChart.tsx (hunk 2 above)
Comment (mapmeld) [reply webui/react/src/components/kit/LineChart.tsx:137]: FixedSizeGrid is going to want height (size of scrollable viewport) and rowHeight (for each chart).
I'll replace settable height with our defaults if that helps

Comment (ClaireNeveu) [reply webui/react/src/components/kit/LineChart.tsx:137]: I think we want the same row height on every chart grid.

**Comment by mapmeld** [comment]
Added diamond shape and chart grid styling:

Screen Shot 2023-01-20 at 1 13 05 PM

Screen Shot 2023-01-20 at 1 14 00 PM

I would want to bring x-axis back to some group consensus.

**Comment by trentwatt** [comment]
> The point indicator still does not match the wireframe and the background color is wrong. Screen Shot 2023-01-20 at 13 35 25 Screen Shot 2023-01-20 at 13 35 55

@ClaireNeveu based on the mockups on [WEB-657] (see below), I think that the <> is reserved for datapoints that are associated with a checkpoint:

image
image


@alexjdetermined can you confirm?

**Comment by ClaireNeveu** [comment]
Interesting. Those wireframes are a bit older than the ones in the ticket. In any case, the diamond is still out of spec with the wireframe. The stroke should be the same as the line and it's too squashed.
Screen Shot 2023-01-24 at 09 40 02

**Comment by mapmeld** [comment]
I think this fixes the diamond. 1.25 width : height

Screen Shot 2023-01-24 at 10 07 27 AM

**Review by ClaireNeveu — APPROVED** [review]
The design looks good. I still think some aspects of the API are a bit wonky but as long as you document the API it's fine.

**Pull request merged**